### DISCUSSION TITLE
Cannot filter records (in table B) selecting before a category (in table A). Each category (A) can have many records in table B

### DISCUSSION THREAD

**Discussion by fablabivrea** [Budibase/budibase#12851] [Help] [answered]
Hi, i'm new with testing this platform....and i have an issue i cannot handle alone....
![Budibase2](https://github.com/Budibase/budibase/assets/7211262/495c4a44-e9c6-4d9f-873e-abd0a833df47)
![Budibase1](https://github.com/Budibase/budibase/assets/7211262/3042dd51-9fd6-41d3-b0cf-33b71d84e9aa)

COntext: the data is stored in a google sheet document
Here main table:
1) POI - Point of interst where each row has a category defined in other table
2) Categories: not more than 20 catogories....es.Outdoor, Museums, Park...and so on
i try to read categories in order to choose one and show only the Points of Interset using a filter. I tried using repeaters for both data set but filter is not working....

Is this due because google sheets aren't  real databases and tables?

**Comment by melohagan** [comment]
Hey @fablabivrea Could you share screenshots of the filters you are using as well please?

**Comment by fablabivrea** [comment]
Here are the screenshots with
1)  action to set of a global variable 
2) filter to to show only the Points of Interest of selected category.  Now in use the formula "IN" but also "EQUAL" or "LIKE" DO NOT WORK.
![Budibase4](https://github.com/Budibase/budibase/assets/7211262/4d8aa918-01ef-4c3f-ac7f-f3e0f1edadff)
![Budibase3](https://github.com/Budibase/budibase/assets/7211262/dfa23eaf-83b8-40ab-a58a-5f9bcc444127)

**Comment by fablabivrea** [comment]
I would like to adopt BUDIBASE in the school where i teach....so i need to handle and resolve this issue. Any help?

**Reply by ConorWebb96** [reply]
Hey @fablabivrea,

I'm not quite sure what your issue is here. I've tried to replicate what I could based off your screenshots, the only difference being instead of categories I've used locations. Categories should work.

Heres my setup within **Google sheets**
<img width="1512" alt="Screenshot 2024-01-25 at 12 31 52" src="https://github.com/Budibase/budibase/assets/126772285/b028bd2a-815e-4cf7-a33e-0808e60a4a5a">
<img width="1512" alt="Screenshot 2024-01-25 at 12 31 59" src="https://github.com/Budibase/budibase/assets/126772285/382beafc-dc14-4aab-b9cf-340968077eee">

Budibase Tables Created syncing in google sheets
<img width="1508" alt="Screenshot 2024-01-25 at 12 30 26" src="https://github.com/Budibase/budibase/assets/126772285/88bfe103-9e6b-4d3b-9618-79d95254bc7c">
<img width="1509" alt="Screenshot 2024-01-25 at 12 30 48" src="https://github.com/Budibase/budibase/assets/126772285/6cd900c1-de4c-41fe-be8c-539c06519ad3">

For locations all i setup a card block component with the data source linked to the locations table. From here I ejected the block and didn't change anything else.
<img width="1511" alt="Screenshot 2024-01-25 at 12 35 06" src="https://github.com/Budibase/budibase/assets/126772285/a258fa9c-1c53-4bac-b470-791ab024f710">

Here is what the component tree for the POI table looks like.
<img width="1317" alt="Screenshot 2024-01-25 at 12 38 22" src="https://github.com/Budibase/budibase/assets/126772285/2808b3b3-8d07-4dca-a22d-28388ffc6e5e">

For the filtering all i've done is the following. 
<img width="1512" alt="Screenshot 2024-01-25 at 12 38 33" src="https://github.com/Budibase/budibase/assets/126772285/87136c3a-b8ac-43b0-a1f5-d102b90bf85e">
What the above does is change the state value based on the location button that is clicked. After the value is updated it will then filter the data shown in that area.

Here is a gif of this whole thing working. I did notice that the time it takes to filter was slower than what I'm used to. But you can definitely get this working with the Google sheets connector.
![20240125_124015](https://github.com/Budibase/budibase/assets/126772285/f55cec36-8f7d-4db2-9210-6b05a02a3bf5)

I can't really see why this wouldn't work for you as well. I hope this helps point you in the right direction.

Some additional questions for you below if the above doesn't help.

- What version of Budibase are you running?
- Are you cloud or self-hosted?
- Can you provide some screenshots of the data contained within your Google sheet tables?
- Can you provide some screenshots of your google sheets themselves?

**Reply by fablabivrea** [reply]
Can i share in private my google sheet? If you want i can also share my test project with you. I'm using budibase shared on budibase ufficial site. So i think last free version

**Comment by fablabivrea** [comment]
![Budibase5](https://github.com/Budibase/budibase/assets/7211262/5efc9417-50cd-4ee7-af4e-2cecfce14c31)
![Budibase6](https://github.com/Budibase/budibase/assets/7211262/afc8025e-fe61-44b1-88d5-95b535c8823c)
![Budibase7](https://github.com/Budibase/budibase/assets/7211262/72a30417-c044-477c-8839-09b50306c7bb)
![Budibase8](https://github.com/Budibase/budibase/assets/7211262/e41c7d54-528a-4b17-ac21-9ec6b72cadc1)

**Reply by ConorWebb96** [reply, marked as answer]
Hey @fablabivrea,

Here is my exported app, I've changed it to use data more similar to yours. You will have to paste in your spread url and re-sync your data as well as select the tables the data is reference from in the data providers.
[Google sheets-export-1706191366265.tar.gz]() ~ removed export link as it has link to spreadsheets (Don't want to get spammed access requests.)

Heres the updated google sheets.
<img width="752" alt="Screenshot 2024-01-25 at 14 07 19" src="https://github.com/Budibase/budibase/assets/126772285/6a4e68e3-7d50-46e2-a14a-d993d6fd8804">
<img width="676" alt="Screenshot 2024-01-25 at 14 07 25" src="https://github.com/Budibase/budibase/assets/126772285/52a4468e-779f-4b1d-8777-4fcabec51673">

Budibase tables after they have been synced.
<img width="1512" alt="Screenshot 2024-01-25 at 14 08 11" src="https://github.com/Budibase/budibase/assets/126772285/d3519c27-2c12-443b-b746-7d02d708a4f1">
<img width="1509" alt="Screenshot 2024-01-25 at 14 08 18" src="https://github.com/Budibase/budibase/assets/126772285/098d14e8-862c-4fbc-a5c2-96ffc39ce9a4">

The working version after clicking a screenshot.
<img width="1511" alt="Screenshot 2024-01-25 at 14 08 43" src="https://github.com/Budibase/budibase/assets/126772285/f6717ebd-efba-4350-8ec3-e6f1c5049f43">

**Comment by fablabivrea** [comment]
Tnx [ConorWebb96](https://github.com/ConorWebb96) for your help. I tried your examples and then i used the same mapping my data....and everithing now is working. You used a form in a way I didn't use before in that way....and, second, you used the button to pass the category as filter to the following data search....

Very useful....i learned a lot from your example !!!
![Budibaseok](https://github.com/Budibase/budibase/assets/7211262/fc8900db-2a8e-4938-b51b-006144b870a5)
